shell/browser/browser_linux.cc

```diff
@@ -113,6 +131,16 @@ bool Browser::RemoveAsDefaultProtocolClient(const std::string& protocol,
   return false;
 }
 
+base::string16 Browser::GetApplicationNameForProtocol(const GURL& url) {
+  std::vector<std::string> argv;
```

Comment:
kinda-nit: for simplicity we can probably just do:
```cpp
const std::string& handler = std::string("x-scheme-handler/") + url.scheme();
argv = {"xdg-mime", "query", "default", handler};
```

Reply:
Good idea. Updated the other similar uses as well.